### PR TITLE
fix:Encrypting blueprint.settings blueprint.plan and pipeline.plan in the database #2868

### LINKED ISSUE #2868: [Feature][framework] Encrypting `blueprint.settings` `blueprint.plan` and `pipeline.plan` in the database
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/incubator-devlake/issues?q=is%3Aissue) and found no similar feature requirement.

### Description

According to the discussion in #2800.
It is totally legit to encrypt those fields in the database since they might contain sensitive information like git repo password/token.

### Use case

enhance the security in case of Database Leaks

### Related issues

#2800

### Are you willing to submit a PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by yuqiangabab**
# Summary

### Does this close any open issues?
Closes #2868

### Screenshots
Include any relevant screenshots here.

### Other Information
Any other information that is important to this PR.

### CODE REVIEW

**Review by e2corporation — COMMENTED** [review]
Hello @yuqiangabab, thank you for the PR. As far as the `config-ui` subpackage requirements for this PR, it doesn't look like there are any valid `package.json` changes for UI, so the `package-lock.json` file should be removed from this PR as it's only been modified as a result of your local development environment.

**Comment by yuqiangabab** [comment]
> Hello @yuqiangabab, thank you for the PR. As far as the `config-ui` subpackage requirements for this PR, it doesn't look like there are any valid `package.json` changes for UI, so the `package-lock.json` file should be removed from this PR as it's only been modified as a result of your local development environment.

Okay, I've dealt with it

File: services/pipeline_helper.go [hunk 1]
```diff
@@ -0,0 +1,208 @@
...
+		return nil, errors.Internal.Wrap(err, "error getting the pipeline from database", errors.AsUserMessage())
+	}
+	return dbPipeline, nil
+}
+
+// parsePipeline
+func parsePipeline(dbPipeline *models.DbPipeline) (*models.Pipeline, error) {
+	pipeline := models.Pipeline{
+		Model: common.Model{
+			ID: dbPipeline.ID,
```
Comment (warren830) [inline services/pipeline_helper.go:150]: should also add CreatedAt and UpdatedAt when you are getting it

File: services/pipeline_helper.go [hunk 2]
```diff
@@ -0,0 +1,208 @@
...
+		Stage:         dbPipeline.Stage,
+	}
+	return &pipeline, nil
+}
+
+// parseDbPipeline
+func parseDbPipeline(pipeline *models.Pipeline) (*models.DbPipeline, error) {
+	dbPipeline := models.DbPipeline{
+		Model: common.Model{
+			ID: pipeline.ID,
```
Comment (warren830) [inline services/pipeline_helper.go:171]: maybe we can directly use `Model: pipeline.Model,`

File: services/blueprint_helper.go [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+	blueprint := models.Blueprint{
+		Name:       DbBlueprint.Name,
+		Mode:       DbBlueprint.Mode,
+		Plan:       []byte(DbBlueprint.Plan),
+		Enable:     DbBlueprint.Enable,
+		CronConfig: DbBlueprint.CronConfig,
+		IsManual:   DbBlueprint.IsManual,
+		Settings:   []byte(DbBlueprint.Settings),
+		Model: common.Model{
+			ID: DbBlueprint.ID,
```
Comment (warren830) [inline services/blueprint_helper.go:95]: maybe we can directly use `Model: pipeline.Model,`

File: services/blueprint_helper.go [hunk 4]
```diff
@@ -0,0 +1,148 @@
...
+		},
+	}
+	return &blueprint, nil
+}
+
+// parseDbBlueprint
+func parseDbBlueprint(blueprint *models.Blueprint) (*models.DbBlueprint, error) {
```
Comment (warren830) [inline services/blueprint_helper.go:102]: we don't need a return err

File: services/blueprint.go [hunk 5]
```diff
@@ -50,7 +50,15 @@ func CreateBlueprint(blueprint *models.Blueprint) error {
...
+		return err
+	}
+	dbBlueprint, err = encryptDbBlueprint(dbBlueprint)
+	if err != nil {
+		return err
+	}
+	err = CreateDbBlueprint(dbBlueprint)
 	if err != nil {
 		return err
 	}
```
Comment (warren830) [inline services/blueprint.go:64]: please add : blueprint.Model = dbBlueprint.Model

**Review by warren830 — APPROVED** [review]
LGTM

**Pull request merged**